src/history/hs_rec.c

```diff
@@ -601,6 +605,21 @@ __wt_hs_insert_updates(
                 WT_ASSERT(session, __txn_visible_id(session, upd->txnid));
 #endif
 
+            /*
+             * Reverse deltas are only supported on 'S' and 'u' value formats.
+             */
+            if (!WT_STREQ(btree->value_format, "S") && !WT_STREQ(btree->value_format, "u"))
+                enable_reverse_modify = false;
+
+            /*
+             * Limit the number of consecutive reverse modifies for standard updates. We want to
+             * ensure we do not store a large chain of reverse modifies as to impact read
+             * performance.
+             */
+            if (upd->type == WT_UPDATE_STANDARD &&
+              total_nb_modifies >= WT_MAX_CONSECUTIVE_REVERSE_MODIFY)
+                enable_reverse_modify = false;
```

Comment:
Should we reenable reverse modify after this one is inserted?

Reply:
You mean if enable_reverse_modify was true before getting in there ?

Reply:
I mean if we have already inserted 10 reverse modifies and we decide to insert a full update. After that, if we still have updates to insert, should they be reverse modifies or full update? From the implementation, I understand they will be full updates, am I correct?

Reply:
Yes that's correct, from the current implementation, they will be full updates once 10 standard updates have been written as reverse modifies. It seems that we should not change the enable_reverse_modify state, I will suggest code changes. Thanks for spotting this !